jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -384,15 +389,26 @@ private void pushContainerConfiguration() {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
 
-    results.containerConfigurationPushResult =
+    results.builtImagesAndContainerConfigurationPushResults =
         executorService.submit(
-            () ->
-                new PushContainerConfigurationStep(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        results.targetRegistryClient.get(),
-                        results.builtImages.get().get(0).get())
-                    .call());
+            () -> {
+              Map<Image, Future<BlobDescriptor>> containerConfigurationPushResults =
+                  new HashMap<>();
+              for (Future<Image> builtImage : results.builtImages.get()) {
+                Future<BlobDescriptor> containerConfigurationPushResult =
+                    executorService.submit(
+                        () ->
+                            new PushContainerConfigurationStep(
+                                    buildContext,
+                                    childProgressDispatcherFactory,
+                                    results.targetRegistryClient.get(),
+                                    builtImage.get())
+                                .call());
+                containerConfigurationPushResults.put(
+                    builtImage.get(), containerConfigurationPushResult);
```

Comment:
So I realized a problem with this. `buildImage` in the loop is a `Future`, so calling `builtImage.get()` means blocking until the image is built. Rather, we don't want the loop to block. We may need to put a `Future<Image>` instead of `Image` as a key?

Reply:
An alternative is to make `PushContainerConfigurationStep` return `Image` and `BlobDescriptor` together, but I think that is more involved.

Reply:
I see what you mean .Since we have already called `builtImage.get()` in the same loop does it really mater to call it again.

Reply:
Note we don't call `buitlImage.get()` inside loop directly, but rather tells `executorService` to spawn a new thread to run the lambda which will later cal lit.

Reply:
Ohh I see the logic now